### PR TITLE
[PYTX] VPDQ Add vpdq test&utils for supporting matching in index

### PR DESCRIPTION
**Pull request by tigerchensh**
Summary
---------

The initiative of this pr is from https://github.com/facebook/ThreatExchange/pull/1097.
get_random_VPDQs and  pdq_hashes_to_VPDQ_features are implemented as test utils to create edge-case scenario for VPDQ matching. 

Move util tests into test_vpdq_faiss.py

Write testcase for VPDQ Index matching 
test_no_match: No match in index. The match percentage (0, 0)-query percentage and index percentage

test_matches: two videos (length 10) each with one unmatched frame. Query video2 with index built from video1 and vice versa. Delete the unmatched frame from video1 and query. Then Delete the unmatched frame from video2 and query. The match percentage should be (90, 90) for both ways of building index, (90, 100) and (100, 100).

test_duplicate_matches: There are video1 (length 10) with one duplicate frame and Video2 (length 9) where matches with all unique frames in video1. The match percentage is (100, 100) because the duplicate one is deduped and not counted.

test_duplicate_video_matches: There are video1 with four frames (each distanced by 85) and video2 which contains the duplicated first two frames in video2. video3 gets the matched frame with video1. Query video3 against index which contains video1 and video2 gives result (100, 100) and (50, 100).

### CODE REVIEW

**Review by Dcallies — APPROVED** [review]
Cases I still want to see (followup PR):
1. Match percentage slightly below default threshold - current no_match test has 0% overlap
2. A empty video
3. A video where every feature is below the quality threshold
4. Demonstrating above/below query->index and index->query thresholds.

File: python-threatexchange/threatexchange/extensions/vpdq/tests/test_vpdq_faiss.py [hunk 1]
```diff
@@ -67,16 +96,107 @@ def test_serialize():
...
+    assert len(res) == 0
+
+
+def test_matches():
+    video1 = pdq_hashes_to_VPDQ_features(
+        [get_similar_hash(get_zero_hash(), 5) for i in range(9)]
+    )
+    video2 = pdq_hashes_to_VPDQ_features(
+        [get_similar_hash(hash.hex, 20) for hash in video1]
+    )
```
[the remark this answers is not in the document]
Comment (tigerchensh) [reply python-threatexchange/threatexchange/extensions/vpdq/tests/test_vpdq_faiss.py:130]: Thanks for the suggestion. Have added these groups to the test.

**Comment by tigerchensh** [comment]
Thanks for the review and suggestions. I have moved some info from the summary into the comments. Maintained groups of test hashes to provide readability.
